cfme/networks/provider/__init__.py

```diff
@@ -232,6 +321,20 @@ def step(self):
         self.prerequisite_view.entities.get_entity(name=self.obj.name).click()
 
 
+@navigator.register(NetworkProvider, 'Edit')
+class Edit(CFMENavigateStep):
+    VIEW = NetworkProviderEditView
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        self.prerequisite_view.entities.get_entity(name=self.obj.name, surf_pages=True).check()
+        try:
+            self.prerequisite_view.toolbar.configuration.item_select('Edit Selected Network Provider')
+        except MoveTargetOutOfBoundsException:
+            # TODO: Remove once fixed 1475303
+            self.prerequisite_view.toolbar.configuration.item_select('Edit Selected Network Provider')
```

Comment:
Perhaps a comment on why just retrying the operation is acceptable.

Reply:
We had broken toolbars at one point when JS decided to be bad on us. Try if it works without the retry code, please :)

Reply:
Sure, I was hoping you knew why (-:   If not no need for a comment.   I can't fault you for those before you not leaving the reason they were dong this.